core/src/consensus/mod.rs

```diff
@@ -215,7 +208,7 @@ pub trait ConsensusEngine<M: Machine>: Sync + Send {
 
     /// Called when proposal block is verified.
     /// Consensus many hold the verified proposal block until it should be imported.
-    fn on_verified_proposal(&self, _header: &Header) {}
+    fn on_verified_proposal(&self, _verified_block_data: &[u8]) {}
```

Comment:
Should we pass the bytes? Is there no abstraction of Block?

Reply:
The caller site has a pre-verified block(`PreverifiedBlock`) and closed block(`ClosedBlock`).

```
let pre_verified_block = ...;
let closed_block = verify(pre_verified_block);
on_verified_proposal(&pre_verified_block.bytes);
```

We can't use the closed block because the closed block's bytes are not the same with the pre-verified block's.
We can't use the pre-verified block because the pre-verified block's type says it is not verified yet. Receiving pre-verified block in `on_verified_proposal` is awkward.

There is also encoded::Block type, I don't know when to use the encoded::Block yet.
I'll check it now.

Reply:
I think `encoded::Block` is enough.

`BlockView` is a wrapper of `rlp<'a>`, and provide getter functions. `encoded::Block` owns `Vec<u8>` and provide `BlockView`'s getters.